File: src/OK.Hookman.UI/static/src/app/_models/event.model.ts
import { SenderModel } from './sender.model';
import { ReceiverModel } from './receiver.model';
import { ActionModel } from './action.model';

export class EventModel {
    constructor(
        public id: number,
        public senderId: number,
        public receiverId: number,
        public actionId: number,
        public method: string,
        public path: string,
        public headers: {[key: string]: string},
        public queryStrings: {[key: string]: string},
        public body: string,
        public retryCount: number,
        public createdDate: Date,
        public updatedDate: Date,
        public sender: SenderModel,
        public receiver: ReceiverModel,
        public action: ActionModel
    ) { }
}

File: src/OK.Hookman.Core/Models/EventModel.cs
using System;
using System.Collections.Generic;

namespace OK.Hookman.Core.Models
{
    public class EventModel
    {
        public int Id { get; set; }
        public int? SenderId { get; set; }
        public int ReceiverId { get; set; }
        public int ActionId { get; set; }
        public string Method { get; set; }
        public string Path { get; set; }
        public IDictionary<string, string> QueryStrings { get; set; }
        public IDictionary<string, string> Headers { get; set; }
        public string Body { get; set; }
        public int RetryCount { get; set; }
        public DateTime CreatedDate { get; set; }
        public DateTime? UpdatedDate { get; set; }

        public SenderModel Sender { get; set; }
        public ReceiverModel Receiver { get; set; }
        public ActionModel Action { get; set; }
    }
}